source/MasterData.Application/MasterDataClient.cs

```
@@ -110,19 +111,17 @@ public async Task<GridAreaOwnerDto> GetGridOwnerForGridAreaCodeAsync(string grid
         return new GridAreaOwnerDto(gridAreaOwner.GridAreaCode, gridAreaOwner.ValidFrom, gridAreaOwner.GridAreaOwnerActorNumber, gridAreaOwner.SequenceNumber);
     }
 
-    public async IAsyncEnumerable<GridAreaOwnerDto> GetAllGridAreaOwnersAsync(
-        [EnumeratorCancellation] CancellationToken cancellationToken)
+    public async Task<ImmutableDictionary<string, ActorNumber>> GetAllGridAreaOwnersAsync(
+        CancellationToken cancellationToken)
     {
-        await foreach (var gridAreaOwner in _gridAreaRepository
-                           .GetAllGridAreaOwnersAsync(cancellationToken)
-                           .ConfigureAwait(false))
+        var builder = ImmutableDictionary.CreateBuilder<string, ActorNumber>();
+        await foreach (var gridAreaOwner in _gridAreaRepository.GetAllGridAreaOwnersAsync(cancellationToken))
         {
-            yield return new GridAreaOwnerDto(
-                gridAreaOwner.GridAreaCode,
-                gridAreaOwner.ValidFrom,
-                gridAreaOwner.GridAreaOwnerActorNumber,
-                gridAreaOwner.SequenceNumber);
+            // TODO (MWO): This should in principle always add, and always successfully add, but we should consider what to do if it fails.
```

Planning to address this comment in this PR or an upcoming one? Perhaps we should raise an exception if we are unable to create a valid set of grid area owners.